src/main/java/uk/gov/pay/directdebit/payments/dao/PaymentViewDao.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.pay.directdebit.payments.dao;
+
+import org.jdbi.v3.core.Jdbi;
+import uk.gov.pay.directdebit.payments.dao.mapper.PaymentViewMapper;
+import uk.gov.pay.directdebit.payments.model.PaymentView;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class PaymentViewDao {
+
+    private final Jdbi jdbi;
+
+    private StringBuilder sb = new StringBuilder(
+            "SELECT ga.external_id as gateway_external_id, " +
+            "pr.external_id AS payment_request_id, " +
+            "pr.amount AS amount, " +
+            "pr.reference AS reference, " +
+            "pr.description AS description, " +
+            "pr.return_url as return_url, " +
+            "pr.created_date as created_date, " +
+            "pa.name AS name, " +
+            "pa.email AS email, " +
+            "tr.state AS state " +
+            "FROM payment_requests pr " +
+            "JOIN payers pa ON pr.id = pa.payment_request_id " +
+            "JOIN transactions tr ON pr.id = tr.payment_request_id " +
+            "JOIN gateway_accounts ga ON ga.id = pr.gateway_account_id " +
+            "WHERE ga.external_id = :gatewayAccountExternalId " +
+            "ORDER BY pr.id DESC OFFSET :offset LIMIT :limit");
+
+    @Inject
+    public PaymentViewDao(Jdbi jdbi) {
```

Comment:
We have been writing dao using jdbi's fluent api (example https://github.com/alphagov/pay-direct-debit-connector/blob/master/src/main/java/uk/gov/pay/directdebit/tokens/dao/TokenDao.java) or any other. It would be good to keep using the same pattern so that we have consistency

Reply:
See above. We've been using declarative API. Here we need fluent API as the arguments are not static, like in the case of TokenDao.